src/main/java/io/vertx/core/net/impl/NetClientImpl.java

```diff
@@ -293,14 +293,5 @@ private void failed(ContextInternal context, Channel ch, Throwable th, Promise<N
     }
     context.emit(th, connectHandler::tryFail);
   }
-
-  @Override
-  protected void finalize() throws Throwable {
-    // Make sure this gets cleaned up if there are no more references to it
-    // so as not to leave connections and resources dangling until the system is shutdown
-    // which could make the JVM run out of file handles.
-    closeFuture.close();
-    super.finalize();
```

Comment:
I noticed you removed the finalizers from `NetClientImpl`, `TCPServerBase`, and `SharedHTTPClient`. Do all these classes have there own `CloseFuture`?

Reply:
yes they do and the new collectable close future replaces those finalizers